discovery_engine_core/core/Cargo.toml

```
@@ -12,6 +12,7 @@ derivative = "2.2.0"
 derive_more = { version = "0.99.17", default-features = false, features = ["display", "from"] }
 displaydoc = "0.2.3"
 figment = { version = "0.10.6", default-features = false, features = ["json"] }
+futures = "0.3.21"
```

we should pick the features, otherwise we will get an executor etc
```suggestion
futures = { version = "0.3.21", default-features = false, features = ["alloc"] }
```